### ISSUE TITLE
Join-concatenate  should maintain timestamps of joined parts

### ISSUE THREAD

**Issue by JeromeNelsonC** [arch1t3cht/Aegisub#55]
Excellent work here, but I wonder if it is too difficult to proportion this detail.
I generate word level timestamps using Whisper_timestamped (https://github.com/linto-ai/whisper-timestamped) but I want to make karaoke style captions, so when I put it into aegisub, i can join many words to make it karaoke. 
But When I use join concatenate, it doesnt maintain my already accurate timestamps, and I have to re-align each word again, which is a lot of work.
So do you have anyway to maintain the timestamps of the different parts of the concatenate, instead of just taking the beginning timestamp of the first part and the ending of the last part?

**Comment by arch1t3cht** [comment]
Can you be more precise in what exactly you want to achieve? What would be an example input, and what output would you expect?

**Comment by JeromeNelsonC** [comment]
So Ill attach two screenshots to show whats is happening
This one shows how it is before I concatenate... notice the timestamps
<img width="1217" alt="Screenshot 2023-06-09 at 14 45 02" src="https://github.com/arch1t3cht/Aegisub/assets/105387403/b6b80223-5513-40c4-98bb-14ce2058ec44">

After I concatenate and commit the spilts in karaoke mode we get this:
<img width="1851" alt="Screenshot 2023-06-09 at 14 45 57" src="https://github.com/arch1t3cht/Aegisub/assets/105387403/3c36cd06-dbc4-460c-8ebf-aa476ab26e2d">
notice that the timestamps for each word dont correspond to the ones I had before.

it should be like this (I manually put them somewhat in place):
<img width="2031" alt="Screenshot 2023-06-09 at 14 52 52" src="https://github.com/arch1t3cht/Aegisub/assets/105387403/5eebff8e-2211-4d40-9cf4-dbb60a0927af">


I hope this helps make it clear.

**Comment by arch1t3cht** [comment]
Wait, so are you concatenating first and *then* applying the automatically generated karaoke splits?
Maybe you just want to use the Subtitle > Join Lines > As Karaoke command instead? This can also be bound to a keyboard shortcut, the command is called `edit/line/join/as_karaoke`.

**Comment by arch1t3cht** [comment]
Sure, there's a lot of ways. The simplest one might be to append a space to every line *before* joining, by running search&replace (Ctrl+H) to replace `$` with ` ` (i.e. a space character) with "Use regular expressions" checked.
If you need more complex behavior, you might be better off with a simple automation script, or a luainterpret snippet.

**Comment by JeromeNelsonC** [comment]
Oh wow .. I'll try the search & replace and let you know... Thanks!

**Comment by JeromeNelsonC** [comment]
Well... I just realized that join/karaoke does about the same this as concatenate... It doesnt preserve the timestamps of the each line. Your search replace worked though...
Anyway to get those timestamps?

**Comment by JeromeNelsonC** [comment]
And btw, is there a shortcut to commit splits in the audio pane?

**Comment by arch1t3cht** [comment]
I'm still not sure what you mean by "preserve the timestamps" here. Doesn't "join as karaoke" set karaoke timing markers that match the timing of the input lines?

And no, there's no shortcut to commit splits. This is likely because it's a button for this specific timing mode and not an action that can be run globally from anywhere. But usually you'd only use that button after manually setting splits. If all you want is to automatically generate splits based on words with timing being guessed based on the length, you could use an automation script.

**Comment by JeromeNelsonC** [comment]
Exactly. Karaoke doesn't use the timestamps from the original lines, it
just guesses it's own timestamps based on the word length (I think)

**Comment by arch1t3cht** [comment]
No, it doesn't guess. It [uses the timestamps of the input lines](https://github.com/arch1t3cht/Aegisub/blob/feature/src/command/edit.cpp#L770,L775) and outputs `\k` tags accordingly.

**Comment by JeromeNelsonC** [comment]
Then what am I doing wrong that its not working?

**Comment by arch1t3cht** [comment]
I don't know, what *are* you doing?

**Comment by JeromeNelsonC** [comment]
I figured out the bug... It only messes up when there is a pause between two lines (a silent time-gap). Because it expects to have one word after the other. Would it be too diffucult to implement a segment with no text when this happens?
I only have an elementary undetrsanding of programing, but I assume there will have to be another 
if loop, 
to check if the ending timestamp of "first" is the same as the beggining timestamp of the "second" and 
then if yes, add a segment "space" with the starts with the end of first and ends with the begging of "second"

I hope it makes sense

**Comment by JeromeNelsonC** [comment]
I guess its hard to change that, right?
